src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -1007,6 +1101,231 @@ static void M()
                 Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "object? x").WithArguments("System.Runtime.CompilerServices.NullableAttribute").WithLocation(5, 16));
         }
 
+        [Fact]
+        public void Tuples()
+        {
+            var source =
+@"public class A
+{
+    public static ((object?, object) _1, object? _2, object _3, ((object?[], object), object?) _4) Nested;
+    public static (object? _1, object _2, object? _3, object _4, object? _5, object _6, object? _7, object _8, object? _9) Long;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            CompileAndVerify(comp, validator: assembly =>
+            {
+                var reader = assembly.GetMetadataReader();
+                var typeDef = GetTypeDefinitionByName(reader, "A");
+                var fieldDefs = typeDef.GetFields().Select(f => reader.GetFieldDefinition(f)).ToArray();
+
+                // Nested tuple
+                var field = fieldDefs.Single(f => reader.StringComparer.Equals(f.Name, "Nested"));
```

Comment:
>StringComparer [](start = 57, length = 14)

Why isn't a simple string comparison appropriate?

Reply:
Simple string comparison would work as well although `MetadataReader.StringComparer` allows comparing a string without resolving the `StringHandle` in the caller.

---
In reply to: [194557597](https://github.com/dotnet/roslyn/pull/27589#discussion_r194557597) [](ancestors = 194557597)